OPHD/UI/ResourceBreakdownPanel.cpp

```diff
@@ -107,46 +107,31 @@ void ResourceBreakdownPanel::resourceCheck()
-
-	r.drawSubImage(mIcons, 5.0f, rect().y() + 5.0f, 64.0f, 16.0f, 16.0f, 16.0f);
-	r.drawSubImage(mIcons, 5.0f, rect().y() + 23.0f, 80.0f, 16.0f, 16.0f, 16.0f);
-	r.drawSubImage(mIcons, 5.0f, rect().y() + 41.0f, 96.0f, 16.0f, 16.0f, 16.0f);
-	r.drawSubImage(mIcons, 5.0f, rect().y() + 59.0f, 112.0f, 16.0f, 16.0f, 16.0f);
-
-	r.drawText(*FONT, "Common Metals", 28.0f, rect().y() + 5.0f, 255, 255, 255);
-	r.drawText(*FONT, "Rare Metals", 28.0f, rect().y() + 23.0f, 255, 255, 255);
-	r.drawText(*FONT, "Common Minerals", 28.0f, rect().y() + 41.0f, 255, 255, 255);
-	r.drawText(*FONT, "Rare Minerals", 28.0f, rect().y() + 59.0f, 255, 255, 255);
-
-	std::string fmt = std::to_string(mPlayerResources->commonMetals());
-	r.drawText(*FONT, fmt, 200.0f - FONT->width(fmt) , rect().y() + 5.0f, 255, 255, 255);
-	
-	fmt = std::to_string(mPlayerResources->rareMetals());
-	r.drawText(*FONT, fmt, 200.0f - FONT->width(fmt), rect().y() + 23.0f, 255, 255, 255);
-
-	fmt = std::to_string(mPlayerResources->commonMinerals());
-	r.drawText(*FONT, fmt, 200.0f - FONT->width(fmt), rect().y() + 41.0f, 255, 255, 255);
-
-	fmt = std::to_string(mPlayerResources->rareMinerals());
-	r.drawText(*FONT, fmt, 200.0f - FONT->width(fmt), rect().y() + 59.0f, 255, 255, 255);
-
-	r.drawSubImage(mIcons, 220.0f, rect().y() + 8.0f,  ICON_SLICE[COMMON_METALS].x(),   ICON_SLICE[COMMON_METALS].y(),   8.0f, 8.0f);
-	r.drawSubImage(mIcons, 220.0f, rect().y() + 26.0f, ICON_SLICE[RARE_METALS].x(),     ICON_SLICE[RARE_METALS].y(),     8.0f, 8.0f);
-	r.drawSubImage(mIcons, 220.0f, rect().y() + 44.0f, ICON_SLICE[COMMON_MINERALS].x(), ICON_SLICE[COMMON_MINERALS].y(), 8.0f, 8.0f);
-	r.drawSubImage(mIcons, 220.0f, rect().y() + 62.0f, ICON_SLICE[RARE_MINERALS].x(),   ICON_SLICE[RARE_MINERALS].y(),   8.0f, 8.0f);
-
-
-	fmt = formatDiff(mPlayerResources->commonMetals() - mPreviousResources.commonMetals());
-	r.drawText(*FONT, fmt, 235.0f, rect().y() + 5.0f, COMMON_MET_COL.red(), COMMON_MET_COL.green(), COMMON_MET_COL.blue());
-
-	fmt = formatDiff(mPlayerResources->commonMinerals() - mPreviousResources.commonMinerals());
-	r.drawText(*FONT, fmt, 235.0f, rect().y() + 23.0f, COMMON_MIN_COL.red(), COMMON_MIN_COL.green(), COMMON_MIN_COL.blue());
-
-	fmt = formatDiff(mPlayerResources->rareMetals() - mPreviousResources.rareMetals());
-	r.drawText(*FONT, fmt, 235.0f, rect().y() + 41.0f, RARE_MET_COL.red(), RARE_MET_COL.green(), RARE_MET_COL.blue());
-
-	fmt = formatDiff(mPlayerResources->rareMinerals() - mPreviousResources.rareMinerals());
-	r.drawText(*FONT, fmt, 235.0f, rect().y() + 59.0f, RARE_MIN_COL.red(), RARE_MIN_COL.green(), RARE_MIN_COL.blue());
+	Renderer& renderer = Utility<Renderer>::get();
+	renderer.drawImageRect(rect(), mSkin);
+
+	const auto commonMetalImageRect = NAS2D::Rectangle{64, 16, 16, 16};
+	const auto rareMetalImageRect = NAS2D::Rectangle{80, 16, 16, 16};
+	const auto commonMineralImageRect = NAS2D::Rectangle{96, 16, 16, 16};
+	const auto rareMineralImageRect = NAS2D::Rectangle{112, 16, 16, 16};
```

Comment:
You can increase readability by pre-declaring the dimensions and y-position:
```cpp
const auto resourceUiYPosition = 16;
const auto resourceUiDimensions = NAS2D::Vector{16,16};
const auto commonMetalImageRect = NAS2D::Rectangle{64, resourceUiYPosition, resourceUiDimensions};
const auto rareMetalImageRect = NAS2D::Rectangle{80, resourceUiYPosition, resourceUiDimensions};
const auto commonMineralImageRect = NAS2D::Rectangle{96, resourceUiYPosition, resourceUiDimensions};
const auto rareMineralImageRect = NAS2D::Rectangle{112, resourceUiYPosition, resourceUiDimensions};
//...
```

Reply:
I had thought about that briefly, but quickly decided I didn't like the style. It tries to reduce repetition, but ends up being more verbose in the end.

I suppose you could calculate the `Rectangle` inside the loop, based on the regular change to `x`. That was another thing I had considered.

I think more long term though, I want those Rectangle values to be read from the data files. They really shouldn't be defined as literals in the source code. That's just a detail we need to deal with for now until we can get the sprite code and data refactored. I'm not quite prepared to tackle that problem yet though.

In terms of where I want to go with this long term, I think having fully formed `Rectangle`s available for use is what it's going to end up like, so trying to build them up from components maybe goes a bit against the style in the long term plan.